streamalert/scheduled_queries/config/services.py

```diff
@@ -20,137 +20,182 @@
 from botocore.exceptions import ProfileNotFound
 
 from streamalert.scheduled_queries.command.processor import CommandProcessor
-from streamalert.scheduled_queries.container.container import ServiceDefinition
+from streamalert.scheduled_queries.config.lambda_conf import get_streamquery_env_vars
+from streamalert.scheduled_queries.container.container import ServiceDefinition, ServiceContainer
 from streamalert.scheduled_queries.handlers.athena import AthenaClient
 from streamalert.scheduled_queries.query_packs.configuration import QueryPackRepository
 from streamalert.scheduled_queries.query_packs.manager import (
     QueryPackExecutionContext,
     QueryPacksManagerFactory,
     QueryParameterGenerator,
 )
-from streamalert.scheduled_queries.state.state_manager import StateManager
+from streamalert.scheduled_queries.state.state_manager import StateManager, StepFunctionStateManager
 from streamalert.scheduled_queries.streamalert.kinesis import KinesisClient
 from streamalert.scheduled_queries.support.clock import Clock
 
 
+# FIXME (Ryxias)
+#   Eventually we should get rid of the ServiceContainer. This pattern isn't really in the spirit
+#   of StreamAlert and is a relic from when StreamQuery was a separately maintained project.
+class ApplicationServices:
+    def __init__(self):
+        # Boot the service container
+        self._service_container = ServiceContainer(get_streamquery_env_vars())
+        configure_container(self._service_container)
+
+    @property
+    def logger(self):
+        return self._service_container.get('logger')
+
+    @property
+    def command_processor(self):
+        return self._service_container.get('command_processor')
+
+    @property
+    def state_manager(self):
+        return self._service_container.get('state_manager')
+
+    @property
+    def clock(self):
+        return self._service_container.get('clock')
+
+    def create_step_function_state_manager(self):
+        return StepFunctionStateManager(
+            self.state_manager,
+            self.logger,
+            self.clock
+        )
+
+
 # pylint: disable=too-many-statements
 def configure_container(container):
```

Comment:
maybe a dumb question - since this seems pretty tied to the `ServiceContainer`, why not just have this be a `classmethod` on that class and not a static function?

Reply:
The ServiceContainer is meant to be a reusable component. It's a write-once, use-anywhere class. Each application can then configure its own container, as each application will have different services and configurations available to it

Reply:
ah got it - thanks for clarifying!